code/model/TimedNotice.php

```diff
@@ -178,6 +184,60 @@ public function canCreate($member = null)
     }
 
     /**
+     * Gets any notices relevant to the present time, context and current users
+     *
+     * @return ArrayList
+     **/
+    public static function get_notices()
+    {
+        // analyse the context this message gets requested
+        $context = self::getContext();
```

Comment:
(facepalm) I should have picked this up. 

About the context passing in: I intentionally didn't pass it in because this would give developers the chance to set it accidentally to 'CMS' while actually on the website and exposing CMS-related messages to a potential attacker. What do you think?